corehq/apps/registration/views.py

```diff
@@ -114,28 +118,44 @@ def register_user(request, domain_type=None):
         return render(request, 'registration/create_new_user.html', context)
 
 
-@transaction.atomic
-@login_required
-def register_domain(request, domain_type=None):
-    domain_type = domain_type or 'commcare'
-    if domain_type not in DOMAIN_TYPES or (not request.couch_user) or request.couch_user.is_commcare_user():
-        raise Http404()
+class RegisterDomainView(View):
```

Comment:
have you looked at inheriting from @TemplateView? That would be more appropriate here